sqldb/sqlite.go

```diff
@@ -0,0 +1,176 @@
+package sqldb
+
+import (
+	"database/sql"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"testing"
+	"time"
+
+	sqlite_migrate "github.com/golang-migrate/migrate/v4/database/sqlite"
+	"github.com/lightningnetwork/lnd/sqldb/sqlc"
+	"github.com/stretchr/testify/require"
+	_ "modernc.org/sqlite" // Register relevant drivers.
+)
+
+const (
+	// sqliteOptionPrefix is the string prefix sqlite uses to set various
+	// options. This is used in the following format:
+	//   * sqliteOptionPrefix || option_name = option_value.
+	sqliteOptionPrefix = "_pragma"
+
+	// sqliteTxLockImmediate is a dsn option used to ensure that write
+	// transactions are started immediately.
+	sqliteTxLockImmediate = "_txlock=immediate"
+
+	// defaultMaxConns is the number of permitted active and idle
+	// connections. We want to limit this so it isn't unlimited. We use the
+	// same value for the number of idle connections as, this can speed up
+	// queries given a new connection doesn't need to be established each
+	// time.
+	defaultMaxConns = 25
+
+	// connIdleLifetime is the amount of time a connection can be idle.
+	connIdleLifetime = 5 * time.Minute
```

Comment:
Will this result in the connection pool shrinking? If so will the pool be repopulated automatically?

Reply:
IIUC, this lets a connection be idle for 5 mins at a time before being garbage colelcted. These "connections" re really just in memory structs (for sqlite) managed by the `database/sql` package. FWIW, these values haven't been tuned a _ton_ other than things like some of the pragma directives we realized we needed.